server/games/game.py

```diff
@@ -391,16 +394,25 @@ def add_game_connection(self, game_connection):
         """
         if game_connection not in self._connections.values():
             return
-        del self._connections[game_connection.player]
-        if game_connection.player:
-            del game_connection.player.game
+
+        player = game_connection.player
+        del self._connections[player]
+
+        if player:
+            del player.game
+
+            if (
+                self.state is GameState.LOBBY and
+                player.id in self._player_options
+            ):
+                del self._player_options[player.id]
+
         await self.check_sim_end()
 
         self._logger.info("Removed game connection %s", game_connection)
 
         def host_left_lobby() -> bool:
```

Comment:
Do we need that function?
1. like i don't see the reason for inner function
2. this is just boolean result for condition, maybe we could just move body to condition

Reply:
Instead of moving the body into the condition directly, I'd make it a variable.
```python
host_left_lobby = (game_connection.player == self.host and self.state is not GameState.LIVE)

if self.state is not GameState.ENDED and (
    len(self._connections) == 0 or host_left_lobby
):
```

Reply:
Yea I'll do that